### PR TITLE
More Customizable Options

### PR DESCRIPTION
**Pull request by AsishRaju**
## More Customization Options ✨

- Change foreground and text color 🎨
- Change opacity/transparency of the foreground 🕹
- Added input for border size #️⃣
- Added 3 more backgroud patterns 🪐
- Added Reset button 🔁
- Choose diffrent border types 🧐
- Now Supports download as JPEG ⏬
- Refactored Codebase 👌🏻

### Side Note 😬

Couldnt fix the allignment for foreground color picker and text color picker , i am bad with css 🙁 and my solution was to use float but author was using flex system so didnt do it.

# ✌🏻

### CODE REVIEW

**Comment by rutikwankhade** [comment]
### Whoa! This is way better now. It's a lot more customizable now. This is a huge PR.
- The border thing is  🔥 
- New patterns are cool.
- The reset button is just perfect.

😍 And thank you for refactoring the codebase. 
I will figure out the alignment. No issue. Well, everything just looks perfect to me. You did a great job. Kudos 🙌

**Comment by AsishRaju** [comment]
@rutikwankhade  thanks to you buddy this was my first ever pr to an open-source project looking forward to more of these 👍🏻

**Comment by praveenscience** [comment]
Alright, this supersedes my idea and fixes #1

File: src/components/ComponentToImg.js [hunk 1]
```diff
@@ -1,36 +1,33 @@
...
+				break;
+			case "JPEG":
+				downloadButton = <button onClick={() => exportComponentAsJPEG(this.componentRef)}>Download</button>;
+		}
+		console.log(downloadButton);
+
+		return (
+			<React.Fragment>
+				<div ref={this.componentRef}>{this.props.children}</div>
+				{/* <button onClick={() => exportComponentAsPNG(this.componentRef)}>Download</button> */}
```
Comment (praveenscience) [inline src/components/ComponentToImg.js:26]: Could have completely removed this.

File: src/components/ComponentToImg.js [hunk 2]
```diff
@@ -1,36 +1,33 @@
...
+			case "PNG":
+				downloadButton = <button onClick={() => exportComponentAsPNG(this.componentRef)}>Download</button>;
+				break;
+			case "JPEG":
+				downloadButton = <button onClick={() => exportComponentAsJPEG(this.componentRef)}>Download</button>;
+		}
+		console.log(downloadButton);
+
+		return (
+			<React.Fragment>
```
Comment (praveenscience) [inline src/components/ComponentToImg.js:24]: We can use `<></>` instead.

**Review by praveenscience — COMMENTED** [review]
Looks good! 😊 But here's my suggestion. Or, I'll just add these as enhancements! @rutikwankhade

File: src/components/ComponentToImg.js (hunk 1 above)
Comment (AsishRaju) [reply src/components/ComponentToImg.js:26]: that's removed in my pr

Comment (praveenscience) [reply src/components/ComponentToImg.js:26]: Super! 😊

**Review by rutikwankhade — APPROVED** [review]

**Pull request merged**